reVX/wind_setbacks/wind_setbacks.py

```diff
@@ -0,0 +1,1386 @@
+        """
+        features = cls._parse_features(features_fpath, crs)
+
+        features['geometry'] = features.buffer(setback)
+        setbacks = [(geom, 1) for geom in features['geometry']]
+
+        return setbacks
+
+    @classmethod
+    def _compute_existing_setbacks(cls, features_fpath, crs, wind_regs,
+                                   tip_height, rotor_diameter):
+        """
+        Compute existing features setbacks
+
+        Parameters
+        ----------
+        features_fpath : str
+            Path to file containg feature to setback from
+        crs : str
+            Coordinate reference system to convert structures geometries into
+        wind_regs : pandas.DataFrame
+            Wind regulations that define setbacks by county
+        tip_height : float
+            Turbine blade tip height in meters
+        rotor_diameter : float
+            Turbine rotor diameter in meters
+
+        Returns
+        -------
+        setbacks : list
+            List of setback geometries
+        """
+        features = cls._parse_features(features_fpath, crs)
+
+        setbacks = []
+        for i in range(len(wind_regs)):
+            cnty = wind_regs.iloc[[i]]
+            setback = cls._get_setback(cnty.iloc[0], tip_height,
+                                       rotor_diameter)
+            if setback is not None:
+                logger.debug('- Computing setbacks for county FIPS {}'
+                             .format(cnty.iloc[0]['FIPS']))
+                tmp = gpd.sjoin(features, cnty,
+                                how='inner', op='intersects')
+                tmp['geometry'] = tmp.buffer(setback)
+
+                setbacks.extend((geom, 1) for geom in tmp['geometry'])
+
+        return setbacks
+
+    def _preflight_check(self, regs_fpath, multiplier):
+        """
+        Run preflight checks on WindSetBack inputs:
+        1) Ensure either a wind regulations .csv is provided, or
+           a setback multiplier
+        2) Ensure wind regulations has county FIPS, map regulations to county
+           geometries from exclusions .h5 file
+        3) Ensure multiplier is a valid entry, either a float or one of
+           {'high': 3, 'moderate': 1.1}
+
+        Parameters
+        ----------
+        regs_fpath : str | None
+            Path to wind regulations .csv file, if None create global
+            setbacks
+        multiplier : int | float | str | None
+            setback multiplier to use if wind regulations are not supplied,
+            if str, must one of {'high': 3, 'moderate': 1.1}
+
+        Returns
+        -------
+        regs : geopands.GeoDataFrame | None
+            Wind Regulations
+        Multiplier : float | None
+            Generic setbacks multiplier
+        """
+        if regs_fpath:
+            if multiplier:
+                msg = ('A wind regulation .csv file was also provided and '
+                       'will be used to determine setback multipliers!')
```

Comment:
while this is clear in the source code, it would not be clear to a user what the csv was provided in addition to (the multiplier).